app/models/adviser.rb

```diff
@@ -48,12 +57,21 @@ def field_order
 
   private
 
+  def check_for_changes
+    @old_firm_id = firm_id_change.first if firm_id_changed?
+  end
```

Comment:
Is there a reason we're setting up an instance variable in an `after_save` callback rather than just calling this directly in the `geocode` callback?